src/platform/silabs/efr32/CHIPCryptoPALPsaEfr32.cpp

```diff
@@ -439,139 +433,79 @@ CHIP_ERROR HKDF_sha::HKDF_SHA256(const uint8_t * secret, const size_t secret_len
+    status = psa_import_key(&attrs, key, key_length, &keyId);
+    VerifyOrExit(status == PSA_SUCCESS, error = CHIP_ERROR_INTERNAL);
 
+    status = psa_mac_compute(keyId, algorithm, message, message_length, out_buffer, out_length, &out_length);
     VerifyOrExit(status == PSA_SUCCESS, error = CHIP_ERROR_INTERNAL);
-    VerifyOrExit(output_length == PSA_HASH_LENGTH(PSA_ALG_SHA_256), error = CHIP_ERROR_INTERNAL);
+
 exit:
-    psa_mac_abort(&operation);
-    psa_reset_key_attributes(&attr);
-    return error;
+    LogPsaError(status);
+    psa_destroy_key(keyId);
+    psa_reset_key_attributes(&attrs);
+
+    return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR HMAC_sha::HMAC_SHA256(const Hmac128KeyHandle & key, const uint8_t * message, size_t message_length, uint8_t * out_buffer,
                                  size_t out_length)
 {
-    return HMAC_SHA256(key.As<Symmetric128BitsKeyByteArray>(), sizeof(Symmetric128BitsKeyByteArray), message, message_length,
-                       out_buffer, out_length);
-}
-
-CHIP_ERROR PBKDF2_sha256::pbkdf2_sha256(const uint8_t * password, size_t plen, const uint8_t * salt, size_t slen,
-                                        unsigned int iteration_count, uint32_t key_length, uint8_t * output)
-{
-    // TODO: replace inlined algorithm with usage of the PSA key derivation API once implemented
-    CHIP_ERROR error          = CHIP_NO_ERROR;
-    psa_status_t status       = PSA_ERROR_BAD_STATE;
-    psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
-    size_t output_length      = 0;
-
-    // Align these buffers on the native data size to speed up the XOR
-    static const size_t hash_size_in_native =
-        ((PSA_HASH_LENGTH(PSA_ALG_SHA_256) + sizeof(unsigned int) - 1) / sizeof(unsigned int));
-    static_assert(hash_size_in_native * sizeof(unsigned int) >= PSA_HASH_LENGTH(PSA_ALG_SHA_256));
-
-    unsigned int md1_buffer[hash_size_in_native];
-    unsigned int work_buffer[hash_size_in_native];
-    uint8_t * md1  = (uint8_t *) md1_buffer;
-    uint8_t * work = (uint8_t *) work_buffer;
-
-    size_t use_len;
-    unsigned char * out_p = output;
-    uint8_t * U1          = (uint8_t *) MemoryCalloc(1, slen + 4);
-
-    VerifyOrExit(U1 != nullptr, error = CHIP_ERROR_NO_MEMORY);
-    VerifyOrExit(password != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(plen > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(salt != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(slen >= kSpake2p_Min_PBKDF_Salt_Length, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(slen <= kSpake2p_Max_PBKDF_Salt_Length, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(key_length > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrExit(output != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(IsBufferNonEmpty(message, message_length), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Why is it illegal to do a HMAC of empty? It is not illegal per RFC and other public API support.